release_dashboard/lib/widgets/create_release_substeps.dart

```diff
@@ -112,30 +153,54 @@ class CreateReleaseSubstepsState extends State<CreateReleaseSubsteps> {
 }
 
 typedef SetReleaseData = void Function(String name, String data);
+typedef ChangeIsEachInputValid = void Function(String name, bool isValid);
 
 /// Captures the input values and updates the corresponding field in [releaseData].
 class InputAsSubstep extends StatelessWidget {
-  const InputAsSubstep({
+  InputAsSubstep({
     Key? key,
-    required this.index,
+    required this.substepName,
     required this.setReleaseData,
     this.hintText,
+    required this.changeIsInputValid,
   }) : super(key: key);
 
-  final int index;
+  final String substepName;
   final SetReleaseData setReleaseData;
   final String? hintText;
+  final ChangeIsEachInputValid changeIsInputValid;
 
   @override
   Widget build(BuildContext context) {
+    late RegExp formRegexValidator;
+    late String validatorErrorMsg;
+
+    Map<String, Object> RegexAndErrorMsg = git(name: substepName).getRegexAndErrorMsg();
```

Comment:
If we're only passing two values, it would be clearer to have a helper class like `GitValidator`, and have it offer `GitValidator.regex` and `GitValidator.hint`

Reply:
Generally, you want to avoid maps where possible when you know what the data is going to be.